test/methods/stream/publish-event.js

```diff
@@ -19,220 +19,214 @@ module.exports = () => {
+				done();
+			};
+			document.addEventListener('oTracking.event', listener);
 
 			const stream = new Stream();
 			stream.publishEvent({ name: 'ready' });
-
-			proclaim.isTrue(listenerStub.calledOnce);
 		});
 
 		it("sets isWithheld to true when a comment is withheld for moderation", (done) => {
 			const listener = (event) => {
+				document.removeEventListener('oTracking.event', listener);
 				proclaim.isTrue(event.detail.isWithheld);
 				done();
 			};
 			document.addEventListener('oTracking.event', listener);
 
 			const stream = new Stream();
-			stream.publishEvent({ name: 'createComment.success', data: {
-				success: {
-					status: 'SYSTEM_WITHHELD'
+			stream.publishEvent({
+				name: 'createComment.success',
+				data: {
+					success: {
+						status: 'SYSTEM_WITHHELD'
+					}
 				}
-			}});
-
-			document.removeEventListener('oTracking.event', listener);
+			});
 		});
 
 		it("doesn't emit oTracking events if it has been disabled", (done) => {
-			const listenerStub = sandbox.stub();
-			const commentsElement = document.getElementById('o-comments-stream');
-
-			document.addEventListener('oTracking.event', (event) => {
+			const oTrackingEventListener = (event) => {
 				if (event.detail.category === 'comment' && event.detail.action === 'ready') {
-					listenerStub();
+					document.removeEventListener('oTracking.event', oTrackingEventListener);
+					proclaim.fail('This event should not have been fired');
 				}
-			});
-
-			commentsElement.addEventListener('oComments.ready', () => {
-				listenerStub();
-
-				window.setTimeout(done, 500);
-
-			});
+			};
+			document.addEventListener('oTracking.event', oTrackingEventListener);
 
 			// Removing the o-component data attribute will prevent auto-initialisation
+			const commentsElement = document.getElementById('o-comments-stream');
 			delete commentsElement.dataset.oComponent;
 
 			const stream = new Stream(commentsElement, {
 				disableOTracking: true
 			});
-
 			stream.publishEvent({ name: 'ready' });
+			document.removeEventListener('oTracking.event', oTrackingEventListener);
 
-			proclaim.isTrue(listenerStub.calledOnce);
+			window.setTimeout(() => {
+				document.removeEventListener('oTracking.event', oTrackingEventListener);
+				done();
+			}, 500);
 		});
 
 		it("only maps 1 event every 100 milliseconds", (done) => {
 			const listenerStub = sandbox.stub();
 			document.addEventListener('oComments.ready', listenerStub);
 
 			const stream = new Stream();
-
 			const interval = window.setInterval(() => {
 				stream.publishEvent({ name: 'ready' });
 			}, 10);
 
 			window.setTimeout(() => {
-				proclaim.isTrue(listenerStub.calledTwice);
 				window.clearInterval(interval);
+				document.removeEventListener('oComments.ready', listenerStub);
+				proclaim.isTrue(listenerStub.calledTwice);
 				done();
 			}, 120);
 		});
 	});
 
 	describe("valid coral error", () => {
 		it("maps coral errors to oComment events", (done) => {
```

Comment:
Same with this one, it's missing an assertion. Otherwise, doesn't it always just pass?

Reply:
This is intentional. In this test we are only expecting the event listener to be invoked. When this happens we call done to tell the framework the test is complete.

If the event listener is not invoked, we never call done and the test fails because of timeout.

Reply:
Got it, so we're just letting it time out 👍

Reply:
Yes, I have tested it by not firing that event and confirmed that the test is failing.